inc/plugins/class-limited-offers.php

```diff
@@ -0,0 +1,334 @@
+use DateTimeZone;
+use Exception;
+
+/**
+ * Class LimitedOffers
+ */
+class LimitedOffers {
+
+	/**
+	 * Active deal.
+	 *
+	 * @var string
+	 */
+	private $active = '';
+
+	/**
+	 * The key for WP Options to disable the dashboard notification.
+	 *
+	 * @var string
+	 */
+	public $wp_option_dismiss_notification_key_base = 'dismiss_themeisle_notice_event_';
+
+	/**
+	 * Offer Links
+	 *
+	 * @var array<string>
+	 */
+	public $offer_metadata = array();
+
+	/**
+	 * LimitedOffers constructor.
+	 */
+	public function __construct() {
+		try {
+			if ( $this->is_deal_active( 'bf' ) ) {
+				$this->activate_bff();
+			}
+		} catch ( Exception $e ) {
+			if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+				error_log( $e->getMessage() ); // phpcs:ignore
+			}
+		}
+	}
+
+	/**
+	 * Load hooks for the dashboard.
+	 *
+	 * @return void
+	 */
+	public function load_dashboard_hooks() {
+		add_filter( 'themeisle_products_deal_priority', array( $this, 'add_priority' ) );
+		add_action( 'admin_notices', array( $this, 'render_dashboard_banner' ) );
+		add_action( 'wp_ajax_dismiss_themeisle_event_notice_otter', array( $this, 'disable_notification_ajax' ) );
+	}
+
+	/**
+	 * Check if we have an active deal.
+	 *
+	 * @return bool True if the deal is active.
+	 */
+	public function is_active() {
+		return ! empty( $this->active );
+	}
+
+	/**
+	 * Activate the Black Friday deal.
+	 *
+	 * @return void
+	 */
+	public function activate_bff() {
+		$this->active = 'bf';
+
+		$this->offer_metadata = array(
+			'bannerUrl'     => OTTER_BLOCKS_URL . 'assets/images/black-friday-banner.png', // TODO: change this based on product.
+			'bannerAlt'     => 'Otter Black Friday Sale',
+			'linkDashboard' => tsdk_utmify( 'https://themeisle.com/plugins/otter-blocks/blackfriday', 'blackfridayltd23', 'dashboard' ), // TODO: change this based on product.
+			'linkGlobal'    => tsdk_utmify( 'https://themeisle.com/plugins/otter-blocks/blackfriday', 'blackfridayltd23', 'globalnotice' ),
+		);
+	}
+
+	/**
+	 * Get the slug of the active deal.
+	 *
+	 * @return string Active deal.
+	 */
+	public function get_active_deal() {
+		return $this->active;
+	}
+
+	/**
+	 * Check if the deal is active with the given slug.
+	 *
+	 * @param string $slug Slug of the deal.
+	 *
+	 * @throws Exception When date is invalid.
+	 */
+	public function is_deal_active( $slug ) {
+
+		if ( empty( $slug ) ) {
+			return false;
+		}
+
+		if ( 'bf' === $slug ) {
+			$start_date = '2023-10-10 00:00:00'; // TODO: Add the correct date.
+			$end_date   = '2023-11-27 23:59:00';
+		} else {
+			return false;
+		}
+
+		return $this->check_date_range( $start_date, $end_date );
+	}
+
+	/**
+	 * Get the remaining time for the deal in a human readable format.
+	 *
+	 * @param string $slug Slug of the deal.
+	 * @return string Remaining time for the deal.
+	 */
+	public function get_remaining_time_for_deal( $slug ) {
+		if ( empty( $slug ) ) {
+			return '';
+		}
+
+		if ( 'bf' === $slug ) {
+			$end_date = '2023-11-27 23:59:00';
```

Comment:
We can set the start, end date is a constant on top so we don't have to reference it in multiple places like this.